--- a/rust/private/rustc.bzl
+++ b/rust/private/rustc.bzl
@@ -1028,6 +1028,8 @@
 
     # Wrapper args first
     process_wrapper_flags = ctx.actions.args()
+    process_wrapper_flags.set_param_file_format("multiline")
+    process_wrapper_flags.use_param_file("@%s", use_always = bool(ctx.executable._process_wrapper))
 
     for build_env_file in build_env_files:
         process_wrapper_flags.add("--env-file", build_env_file)
